src/sched.rs

```diff
@@ -1,204 +1,127 @@
-
-    #[inline]
-    pub fn set_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur | (1u32 << bit)
-    }
-
-    #[inline]
-    pub fn clear_cpu_mask_flag(cur: CpuMask, bit: usize) -> CpuMask {
-        cur & !(1u32 << bit)
-    }
-}
 
 pub type CloneCb<'a> = Box<FnMut() -> isize + 'a>;
 
 // A single CPU mask word
-pub type CpuMask = c_ulong;
+type CpuMask = c_ulong;
 
 // Structure representing the CPU set to apply
 #[repr(C)]
 #[derive(Clone, Copy)]
 pub struct CpuSet {
-    cpu_mask: [CpuMask; cpuset_attribs::CPU_SETSIZE/cpuset_attribs::CPU_MASK_BITS]
+    cpu_mask: [CpuMask; CPU_SETSIZE as usize / cpuset_attribs::CPU_MASK_BITS],
 }
 
 impl CpuSet {
     pub fn new() -> CpuSet {
-        CpuSet {
-            cpu_mask: unsafe { mem::zeroed() }
-        }
+        CpuSet { cpu_mask: unsafe { mem::zeroed() } }
     }
 
     pub fn set(&mut self, field: usize) {
         let word = field / cpuset_attribs::CPU_MASK_BITS;
         let bit = field % cpuset_attribs::CPU_MASK_BITS;
 
-        self.cpu_mask[word] = cpuset_attribs::set_cpu_mask_flag(self.cpu_mask[word], bit);
+        self.cpu_mask[word] = set_cpu_mask_flag(self.cpu_mask[word], bit);
     }
 
     pub fn unset(&mut self, field: usize) {
         let word = field / cpuset_attribs::CPU_MASK_BITS;
         let bit = field % cpuset_attribs::CPU_MASK_BITS;
 
-        self.cpu_mask[word] = cpuset_attribs::clear_cpu_mask_flag(self.cpu_mask[word], bit);
+        self.cpu_mask[word] = clear_cpu_mask_flag(self.cpu_mask[word], bit);
     }
 }
 
 mod ffi {
-    use libc::{c_void, c_int, pid_t, size_t};
-    use super::CpuSet;
+    use libc::{c_void, c_int};
 
-    pub type CloneCb = extern "C" fn (data: *const super::CloneCb) -> c_int;
+    pub type CloneCb = extern "C" fn(data: *const super::CloneCb) -> c_int;
 
     // We cannot give a proper #[repr(C)] to super::CloneCb
     #[allow(improper_ctypes)]
-    extern {
+    extern "C" {
         // create a child process
         // doc: http://man7.org/linux/man-pages/man2/clone.2.html
-        pub fn clone(
-            cb: *const CloneCb,
-            child_stack: *mut c_void,
-            flags: c_int,
-            arg: *mut super::CloneCb,
-            ...) -> c_int;
-
-        // disassociate parts of the process execution context
-        // doc: http://man7.org/linux/man-pages/man2/unshare.2.html
-        pub fn unshare(flags: c_int) -> c_int;
-
-        // reassociate thread with a namespace
-        // doc: http://man7.org/linux/man-pages/man2/setns.2.html
-        pub fn setns(fd: c_int, nstype: c_int) -> c_int;
-
-        // Set the current CPU set that a task is allowed to run on
-        pub fn sched_setaffinity(__pid: pid_t, __cpusetsize: size_t, __cpuset: *const CpuSet) -> c_int;
+        pub fn clone(cb: *const CloneCb,
+                     child_stack: *mut c_void,
+                     flags: c_int,
+                     arg: *mut super::CloneCb,
+                     ...)
+                     -> c_int;
```

Comment:
Yeah there's some fanciness going on here to allow passing a Rust closure as the function to run in the new process. There's also #360 which is concerning.